drivers/gles3/storage/config.cpp

```diff
@@ -84,7 +84,11 @@ Config::Config() {
 	glGetInteger64v(GL_MAX_TEXTURE_IMAGE_UNITS, &max_texture_image_units);
 	glGetInteger64v(GL_MAX_TEXTURE_SIZE, &max_texture_size);
 	glGetInteger64v(GL_MAX_UNIFORM_BLOCK_SIZE, &max_uniform_buffer_size);
-	glGetInteger64v(GL_MAX_VIEWPORT_DIMS, max_viewport_size);
+
+	GLsizei max_vp[2] = {};
+	glGetIntegerv(GL_MAX_VIEWPORT_DIMS, max_vp);
+	max_viewport_size[0] = max_vp[0];
+	max_viewport_size[1] = max_vp[1];
```

Comment:
For the reference, it is defined as 32-bit int in ANGLE (which is usually used for WebGL):

https://github.com/google/angle/blob/4d47c73a627a2cb65d943bd764b21e9a33ca4faf/src/libANGLE/queryutils.cpp#L3303-L3308

But the same is true for the other values checked here, except `GL_MAX_UNIFORM_BLOCK_SIZE`.

Reply:
Then I would suggest to apply this patch file:
```diff
diff --git a/drivers/gles3/storage/config.cpp b/drivers/gles3/storage/config.cpp
index 6b5e227782..40dd340ecf 100644
--- a/drivers/gles3/storage/config.cpp
+++ b/drivers/gles3/storage/config.cpp
@@ -80,11 +80,11 @@ Config::Config() {
 		rgtc_supported = extensions.has("GL_EXT_texture_compression_rgtc") || extensions.has("GL_ARB_texture_compression_rgtc") || extensions.has("EXT_texture_compression_rgtc");
 	}
 
-	glGetInteger64v(GL_MAX_VERTEX_TEXTURE_IMAGE_UNITS, &max_vertex_texture_image_units);
-	glGetInteger64v(GL_MAX_TEXTURE_IMAGE_UNITS, &max_texture_image_units);
-	glGetInteger64v(GL_MAX_TEXTURE_SIZE, &max_texture_size);
+	glGetIntegerv(GL_MAX_VERTEX_TEXTURE_IMAGE_UNITS, &max_vertex_texture_image_units);
+	glGetIntegerv(GL_MAX_TEXTURE_IMAGE_UNITS, &max_texture_image_units);
+	glGetIntegerv(GL_MAX_TEXTURE_SIZE, &max_texture_size);
+	glGetIntegerv(GL_MAX_VIEWPORT_DIMS, max_viewport_size);
 	glGetInteger64v(GL_MAX_UNIFORM_BLOCK_SIZE, &max_uniform_buffer_size);
-	glGetInteger64v(GL_MAX_VIEWPORT_DIMS, max_viewport_size);
 
 	support_anisotropic_filter = extensions.has("GL_EXT_texture_filter_anisotropic");
 	if (support_anisotropic_filter) {
diff --git a/drivers/gles3/storage/config.h b/drivers/gles3/storage/config.h
index c3ab65f0bc..026707aa88 100644
--- a/drivers/gles3/storage/config.h
+++ b/drivers/gles3/storage/config.h
@@ -57,14 +57,15 @@ public:
 	bool use_nearest_mip_filter = false;
 	bool use_depth_prepass = true;
 
-	int64_t max_vertex_texture_image_units = 0;
-	int64_t max_texture_image_units = 0;
-	int64_t max_texture_size = 0;
-	int64_t max_viewport_size[2] = { 0, 0 };
+	int32_t max_vertex_texture_image_units = 0;
+	int32_t max_texture_image_units = 0;
+	int32_t max_texture_size = 0;
+	int32_t max_viewport_size[2] = { 0, 0 };
 	int64_t max_uniform_buffer_size = 0;
-	int64_t max_renderable_elements = 0;
-	int64_t max_renderable_lights = 0;
-	int64_t max_lights_per_object = 0;
+
+	int32_t max_renderable_elements = 0;
+	int32_t max_renderable_lights = 0;
+	int32_t max_lights_per_object = 0;
 
 	bool generate_wireframes = false;
```

Reply:
Sure, I can change all variables and related query functions to match OpenGL specification. But won't this be too much of a change, given the other parts of the engine that will be referenced?

Interestingly, GL_MAX_VIEWPORT_DIMS is saved here only to be later mapped to Size2i in `Size2i Utilities::get_maximum_viewport_size()`:

https://github.com/godotengine/godot/blob/5833f597865c773fae3ee09fc4e31d4a243f812d/drivers/gles3/storage/utilities.cpp#L470

Size2i = Vector2i = 2 x 32 bit 💣 

In Vulcan we query for `uint32_t maxViewportDimensions[2];` and in D3D12 simply returning "16384" 🤣 

https://github.com/godotengine/godot/blob/5833f597865c773fae3ee09fc4e31d4a243f812d/drivers/d3d12/rendering_device_driver_d3d12.cpp#L6174-L6176

Anyway, question for you guys, is it safe to change rest of those variables to GLint 32 bit?

I don't know why GL_MAX_UNIFORM_BLOCK_SIZE is different, given that it is mostly 16384-65536:

```
GL_MAX_UNIFORM_BLOCK_SIZE
    params returns one value, the maximum size in basic machine units of a uniform block. The value must be at least 16384. See glUniformBlockBinding. 
```

Reply:
> Anyway, question for you guys, is it safe to change rest of those variables to GLint 32 bit?

We need to use the function that the spec requires. Some will require the 64 but variant while others won't. Don't worry about the rest of the engine